x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlQueryResponse.java

```diff
@@ -242,7 +242,7 @@ private static Object valueAt(String dataType, Block block, int offset, BytesRef
             case "boolean" -> ((BooleanBlock) block).getBoolean(offset);
             case "version" -> new Version(((BytesRefBlock) block).getBytesRef(offset, scratch)).toString();
             case "unsupported" -> UnsupportedValueSource.UNSUPPORTED_OUTPUT;
-            default -> throw EsqlUnsupportedOperationException.unsupportedDataType(dataType);
+            default -> throw ExceptionUtils.deadCode(ExceptionUtils.unsupportedDataType(dataType));
```

Comment:
I am not entirely happy with how this looks with `ExceptionUtils::deadCode`

`ExceptionUtils::deadCode` makes sure that this code results in an error if assertions are enabled. However, as far as I can tell we want this behavior whenever we throw an `EsqlUnsupportedOperationException`, making this needlessly verbose.

We could also make it, so that an assertion error is thrown whenever an `UnsupportedOperationException` is constructed. This would ensure the correct behavior, but I'm not sure if baking an assertion into an exception might lead to confusion/unexpected behavior later down the roead.

Reply:
Yeah, every usage of them in the `esql` package looks like a bug to me, resp. like code that is not intended to ever be reached.

Removed the `deadCode` method, using an `assert false` in the constructors now, as you suggested.